executor/show_stats.go

```diff
@@ -139,14 +139,20 @@ func (e *ShowExec) appendTableForStatsBuckets(dbName, tblName, partitionName str
 	if statsTbl.Pseudo {
 		return nil
 	}
+	colNameToType := make(map[string]byte)
 	for _, col := range statsTbl.Columns {
-		err := e.bucketsToRows(dbName, tblName, partitionName, col.Info.Name.O, 0, col.Histogram)
+		err := e.bucketsToRows(dbName, tblName, partitionName, col.Info.Name.O, 0, col.Histogram, nil)
 		if err != nil {
 			return errors.Trace(err)
 		}
+		colNameToType[col.Info.Name.O] = col.Histogram.Tp.Tp
 	}
 	for _, idx := range statsTbl.Indices {
-		err := e.bucketsToRows(dbName, tblName, partitionName, idx.Info.Name.O, len(idx.Info.Columns), idx.Histogram)
+		var idxColumnType []byte
```

Comment:
Pre-allocate it.

Reply:
I think we need a new array for each index.